### DISCUSSION TITLE
Problem Building from source.

### DISCUSSION THREAD

**Discussion by Mhsh** [emqx/emqx#5530] [Q&A] [answered]
Hi,
I am trying to install emqx from source and followed the steps mentioned on "https://github.com/emqx/emqx". I am getting below error during "make" command. Any help would be appreciated. I am new to linux.

Cloning into '/home/maahi333/emqx/emqx/_build/default/lib/quicer/submodules/msquic/submodules/wil'...
Submodule path 'submodules/msquic/submodules/googletest': checked out 'e8512bc38c4c0060858c3306b0660a3f126aee30'
Submodule path 'submodules/msquic/submodules/openssl': checked out 'a6e9d76db343605dae9b59d71d2811b195ae7434'
Submodule 'boringssl' (https://boringssl.googlesource.com/boringssl) registered for path 'submodules/msquic/submodules/openssl/boringssl'
Submodule 'krb5' (https://github.com/krb5/krb5) registered for path 'submodules/msquic/submodules/openssl/krb5'
Submodule 'pyca.cryptography' (https://github.com/pyca/cryptography.git) registered for path 'submodules/msquic/submodules/openssl/pyca-cryptography'
Cloning into '/home/maahi333/emqx/emqx/_build/default/lib/quicer/submodules/msquic/submodules/openssl/boringssl'...
Cloning into '/home/maahi333/emqx/emqx/_build/default/lib/quicer/submodules/msquic/submodules/openssl/krb5'...
Cloning into '/home/maahi333/emqx/emqx/_build/default/lib/quicer/submodules/msquic/submodules/openssl/pyca-cryptography'...
Submodule path 'submodules/msquic/submodules/openssl/boringssl': checked out '2070f8ad9151dc8f3a73bffaa146b5e6937a583f'
Submodule path 'submodules/msquic/submodules/openssl/krb5': checked out '890ca2f401924cdcb88f493950b04700bbe52db3'
Submodule path 'submodules/msquic/submodules/openssl/pyca-cryptography': checked out '09403100de2f6f1cdd0d484dcb8e620f1c335c8f'
Submodule path 'submodules/msquic/submodules/wil': checked out '86353933d5fba0caf61503c8b09e10e182dccdfb'
CMake Error: The source directory "/home/maahi333/emqx/emqx/_build/default/lib/quicer/c_build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.

**Comment by qzhuyan** [comment, marked as answer]
You can either 
1. build without quic support
```
BUILD_WITHOUT_QUIC=1 make
```
OR

2. ensure to have the following dependencies 
- cmake 3.16+
- libatomic
